src/backend/executor/nodeDML.c

```diff
@@ -183,6 +191,25 @@ ExecDML(DMLState *node)
 				   PLANGEN_OPTIMIZER /* Plan origin */,
 				   isUpdate);
 	}
+	else /* DML_UPDATE */
```

Comment:
may be put this in first, as in 7X?
```suggestion
	if (!AttributeNumberIsValid(plannode->actionColIdx))
	...
	else if (DML_INSERT == action)
	...
	else /* DML_DELETE */
	...
```

Reply:
in 7X, this logic is inside the update-only block. whereas the DML node handles other operations as well. Therefore, I see no point in changing the order.

Reply:
`action == -1` and `!AttributeNumberIsValid(plannode->actionColIdx)` are equivalent conditions

Reply:
Okay, I get it.